python/tvm/relax/frontend/pytorch_fx.py

```diff
@@ -0,0 +1,758 @@
+            )
+            bias = self.params[module.bias] = reshaped_bias
+
+        return self.block_builder.emit(relax.op.add(dense, bias))
+
+    def _relu(self, node: fx.node.Node) -> relax.Var:
+        x = self.env[node.args[0]]
+        return self.block_builder.emit(relax.op.nn.relu(x))
+
+    def _max_pool2d(self, node: fx.node.Node) -> relax.Var:
+        x = self.env[node.args[0]]
+        if node.target in self.named_modules:
+            module = self.named_modules[node.target]
+            kernel = module.kernel_size
+            stride = module.stride
+            padding = module.padding
+            dilation = module.dilation
+        else:
+            nargs = len(node.args)
+            kernel = node.args[1] if nargs > 1 else node.kwargs["kernel_size"]
+            stride = node.args[2] if nargs > 2 else node.kwargs["stride"]
+            padding = node.args[3] if nargs > 3 else node.kwargs["padding"]
+            dilation = node.args[4] if nargs > 4 else node.kwargs["dilation"]
+
+        kernel = kernel if isinstance(kernel, (list, tuple)) else (kernel, kernel)
+        stride = kernel if stride is None else stride
+        stride = stride if isinstance(stride, (list, tuple)) else (stride, stride)
+        padding = (
+            padding if isinstance(padding, (list, tuple)) else (padding, padding, padding, padding)
+        )
+        dilation = dilation if isinstance(dilation, (list, tuple)) else (dilation, dilation)
+
+        return self.block_builder.emit(
+            relax.op.nn.max_pool2d(
+                x,
+                pool_size=kernel,
+                strides=stride,
+                padding=padding,
+                dilation=dilation,
+                layout="NCHW",
+            )
+        )
+
+    def _embedding(self, node: fx.node.Node) -> relax.Var:
+        x = self.env[node.args[0]]
+        module = self.named_modules[node.target]
+        weight = self.params[module.weight]
+        x = self.block_builder.emit(relax.op.astype(x, "int32"))
+        return self.block_builder.emit(relax.op.take(weight, x, axis=0))
+
+    def _adaptive_avg_pool2d(self, node: fx.node.Node) -> relax.Var:
+        module = self.named_modules[node.target]
+        x = self.env[node.args[0]]
+
+        return self.block_builder.emit(
+            relax.op.nn.adaptive_avg_pool2d(x, module.output_size, layout="NCHW")
+        )
+
+    def _flatten(self, node: fx.node.Node) -> relax.Var:
+        x = self.env[node.args[0]]
+        if node.target in self.named_modules:
+            module = self.named_modules[node.target]
+            start_dim = module.start_dim
+            end_dim = module.end_dim
+        else:
+            start_dim = node.args[1] if len(node.args) >= 2 else 0
+            end_dim = node.args[2] if len(node.args) == 3 else -1
+        assert start_dim == 1 and end_dim == -1
+        return self.block_builder.emit(relax.op.flatten(x))
```

Comment:
Our flatten op flattens all dimensions into single one, which is different from PyTorch’s one. Flatten is essentially a reshape operation, so perhaps we will remove our flatten op in the future as it doesn’t differ from reshape. For this case, I think we can only use reshape because our flatten op firstly doesn’t support start/end dim, and secondly cannot do partial flatten.